### ISSUE TITLE
Quickeebooks::Windows::Service::Job#list CustomFieldEnable error

### ISSUE THREAD

**Issue by patrickreiner** [ruckus/quickeebooks#70]
The following line:

```
filters << Quickeebooks::Shared::Service::Filter.new(:boolean, :field => 'CustomFieldEnable', :value => true)
```

causes the following error:

```
IntuitRequestException: cvc-complex-type.2.4.a: Invalid content was found starting with element 'CustomFieldEnable'. One of
'{
  "http://www.intuit.com/sb/cdm/v2":CustomFieldQueryParam,
  "http://www.intuit.com/sb/cdm/v2":CustomFieldFilter,
  "http://www.intuit.com/sb/cdm/v2":CustomFieldDefinitionIdSet,
  "http://www.intuit.com/sb/cdm/v2":ListIdSet,
  "http://www.intuit.com/sb/cdm/v2":StartCreatedTMS,
  "http://www.intuit.com/sb/cdm/v2":EndCreatedTMS,
  "http://www.intuit.com/sb/cdm/v2":StartModifiedTMS,
  "http://www.intuit.com/sb/cdm/v2":EndModifiedTMS,
  "http://www.intuit.com/sb/cdm/v2":CustomerIdSet,
  "http://www.intuit.com/sb/cdm/v2":IncludeFinancialIndicator
}' is expected.
```

I simply commented it out to avoid the issue because I am not using custom fields on jobs.

**Comment by ruckus** [comment]
What is your full call to `list` ? I'm thinking you might be specifying arguments and its order issue (Intuit is a stickler about the order of arguments in the XML).

When I call `Service::Job#list` with no args it works:

``` ruby
s = Quickeebooks::Windows::Service::Job.new
s.access_token = token
s.realm_id = realm_id
s.list
```

**Comment by patrickreiner** [comment]
```
filters = [StartCreatedTMS :EQUALS: 2013-06-06T17:13:04, EndCreatedTMS :EQUALS: 2025-01-01T00:00:00]
list(filters, 1, 999, nil, {}).entries

=>

IntuitRequestException: cvc-complex-type.2.4.a: Invalid content was found starting with element 'CustomFieldEnable'. One of '{
  "http://www.intuit.com/sb/cdm/v2":CustomerIdSet,
  "http://www.intuit.com/sb/cdm/v2":IncludeFinancialIndicator
}' is expected.
```

**Comment by ruckus** [comment]
Ugh, I see whats happening. Yep, Intuit is complaining that the auto-supplied filter (CustomFieldEnable) is coming AFTER the Start/End-CreatedTMS ones. If I pre-pend it to the filters array then its OK.

The problem then is that we cannot just auto-prepend it because the user could be supplying other filters which themselves need to come before CustomFieldEnable.

Its hideous but what I think I'll have to do is maintain an ordered array of all the filters for `Job` and will need to inspect what the the user is giving us and re-order the filters for the API call accordingly.

**Comment by ruckus** [comment]
I just released `0.1.10` which should handle this issue transparently - any filters given to `Job#list` will be re-ordered as needed.

I only implemented the required `FILTER_ORDER` in `Job` service but it will be trivial to add it for other service objects as needed.
